src/AiryFunctions/cairy.jl

```diff
@@ -307,18 +301,26 @@ airy_large_argument_cutoff(z::ComplexOrReal{Float32}) = abs(z) > 4
 
 # valid in 0 <= angle(z) <= pi
 # use conjugation for bottom half plane
-airyai_large_args(z::Complex{T}) where T = airyaix_large_args(z) .* exp(-2/3 * z * sqrt(z))
+function airyai_large_args(z::Complex{T}) where T
+    out = airyaix_large_args(z)
+    return isfinite(z) ? out .* exp(-T(2/3) * z * sqrt(z)) : out
+end
 
 function airybi_large_args(z::Complex{T}) where T
-    if imag(z) < zero(T)
-        out = conj.(airybix_large_args(conj(z)))
-    else
-        out = airybix_large_args(z)
-    end
-    return out .* exp(T(2/3) * z * sqrt(z))
+    out = airybix_large_args(z)
+    return isfinite(z) ? out .* exp(T(2/3) * z * sqrt(z)) : out
 end
 
 @inline function airyaix_large_args(z::Complex{T}) where T
+    if ~isfinite(z)
```

Comment:
why `~isfinite`?

Reply:
I must have forgot about the `isinf` command :man_shrugging:

Reply:
no you probably do want to use isfinite since nan's exist. I just forgot that `~` does the same as `!`.

Reply:
Ugh ya you are right I had it at top level which would check for NaN. I need to make an explicit NaN check at the top now because the else branch is where the NaNs will end up and will repeatedly call the airy. Need to just check first to avoid any NaNs going through the routines